### DISCUSSION TITLE
Manage two pyproject.toml and local dependency

### DISCUSSION THREAD

**Discussion by mihalikv** [astral-sh/rye#320] [Q&A]
I have the following structure of the project(with old requirements.txt managed by hand and installed by pipy):
**core_package** -> standalone project, with separate git and own requirements.txt(most of the packages)
**project_specific_package** -> project that needs core_package to live, and have own requirements.txt i.e. 
```
-r core_package/requirements.txt
mssql==1.0.1
```

I can add requirements.txt to **core_package** and **project_specific_package** when developing these things.  
The project structure is:
```
project_specific_package/
     .git
     requirements.txt
     pyproject.toml - in the future with Rye
     core_package/
         .git
         requirements.txt
         pyproject.toml - in the future with Rye
```

Is it possible somehow create just one virtualenv in root of **project_specific_package**? and when I run rye sync inside _core_package/_ it will generate _core_package/requirements.lock_ just from _core_package/pyproject.toml_

I tried to use following pyproject.toml:
```
[project]
name = "project_specific_package"
version = "0.1.0"
description = "Add a short description here"
authors = [
    { name = "", email = "" }
]
dependencies = [
    "core_package @ file:///home/user/project_specific_package/core_package",
]
readme = "README.md"
requires-python = ">= 3.8"

[build-system]
requires = ["setuptools>=61.0"]
build-backend = "setuptools.build_meta"

[tool.rye]
managed = true

```
And when I run `rye sync` it works like expected it will generate requirements.lock with all dependencies from core_package and it will install it to virtualenv. What if I want to add package to **core_package** ? If I navigate to **core_package** and run `rye add` it just won't work. 

One more experience with pip-tools, that I also tried and I am in beginning of using it. I am managing dependencies only in **core_package** and when I want to generate core dependencies I simply run 
`pip-compile --resolver=backtracking --upgrade -o sbcore/requirements.txt sbcore/pyproject.toml`. Is it maybe some option to define source of pyproject.toml with some switch ?

Thanks.

**Comment by mihalikv** [comment]
So my new observation is, that I can use 
```
[tool.rye.workspace]
members=["core_package"]
```
to have only one virtualenv as I need. 

But I still have problem to add package to `core_package/pyproject.toml` with `rye add`

**Reply by bluss** [reply]
What's the error when you try to add a dependency? Sounds like a bug should be filed if you can provide the info
